### PR TITLE
Bug fixes on Subscription termination date format [Fixes Issues #9186]

### PR DESCRIPTION
**Pull request by tim1234ltp**
Fixes https://github.com/HabitRPG/habitica/issues/9186

### Changes
Created a computed value of Subscription termination date which is formatted according to user preference.
The client will show this computed value instead of using a filter on raw value.

### Result (sorry for the non-english strings)
default

### CODE REVIEW

**Comment by Alys** [comment]
@tim1234ltp Regarding the test failure on this PR (from your comment at https://github.com/HabitRPG/habitica/issues/9186#issuecomment-340328253), the same failure occurs when I run the test locally on a copy of `develop`, so this failure isn't coming from your code. I'm commenting here rather than in the issue so that we see this when we're reviewing your PR.

$ mocha test/api/v3/unit/libs/payments/group-plans/group-payments-create.test.js --require test/helpers/start-server.js 
```
....
  27 passing (9s)
  2 pending
  1 failing

  1) Purchasing a group plan for group adds months to members with existing recurring subscription (Amazon):
     AssertionError: expected 4.037242907706093 to be within 3..4
      at _callee16$ (test/api/v3/unit/libs/payments/group-plans/group-payments-create.test.js:478:55)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at GeneratorFunctionPrototype.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:303:22)
      at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at GeneratorFunctionPrototype.tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at PromiseSpawn._promiseFulfilled (node_modules/bluebird/js/release/generators.js:97:49)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:574:26)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)
```

**Comment by tim1234ltp** [comment]
@Alys Thanks for the update. I would leave this PR open and wait for the review.

File: website/client/components/settings/subscription.vue [hunk 1]
```diff
@@ -237,6 +237,10 @@ export default {
         amount: this.numberOfMysticHourglasses,
       };
     },
+    dateTerminated () {
+      if (!this.user.preferences || !this.user.preferences.dateFormat) return moment(this.user.purchased.plan.dateTerminated);
```
Comment (paglias) [inline website/client/components/settings/subscription.vue:241]: this should return a string not a moment object, calling `.format()` should be enough

**Review by paglias — CHANGES_REQUESTED** [review]
Other than the comment I left this looks good, sorry for the delay!

**Comment by tim1234ltp** [comment]
I will try to fix the PR this weekend.

**Comment by tim1234ltp** [comment]
Oops when I updated my repo with the newest one, tons of commit just generated...
Please kindly ignore them.

**Comment by Alys** [comment]
Hi @tim1234ltp! These things happen. :) We will need to have the extra commits removed so that we can accurately review your change, but if you're not sure how to do that, just post here and we'll give you some advice!

**Comment by tim1234ltp** [comment]
@Alys Which commits? Those not made by me? Should I just set up a clean repo and start from that?

**Comment by paglias** [comment]
@tim1234ltp it looks like a lot of extraneous commits got included so it's probably a good idea to start a new clean branch forked from `develop` and re-apply your changes there

**Comment by tim1234ltp** [comment]
I will start a new PR with clean branch and close this PR.

**Pull request closed without merging**